Basic/APIDiorama/Assets/UseCaseSamples/NetvarVsRpc/Scripts/ColorManager.cs

```diff
@@ -0,0 +1,88 @@
+using UnityEngine;
+
+namespace Unity.Netcode.Samples.APIDiorama
+{
+    /// <summary>
+    /// Manages the color of a Networked object
+    /// </summary>
+    public class ColorManager : NetworkBehaviour
+    {
+        [SerializeField]
+        bool m_UseNetworkVariableForColor;
+
+        NetworkVariable<Color32> m_NetworkedColor = new NetworkVariable<Color32>();
+        Material m_Material;
+
+        void Awake()
+        {
+            m_Material = GetComponent<Renderer>().material;
+        }
+
+        public override void OnNetworkSpawn()
+        {
+            base.OnNetworkSpawn();
+            if (IsClient)
+            {
+                if (m_UseNetworkVariableForColor)
+                {
+                    OnClientColorChanged(m_Material.color, m_NetworkedColor.Value);
+                    m_NetworkedColor.OnValueChanged += OnClientColorChanged;
+                }
+            }
+        }
+
+        public override void OnNetworkDespawn()
+        {
+            base.OnNetworkDespawn();
+            if (IsClient)
+            {
+                if (m_UseNetworkVariableForColor)
+                {
+                    m_NetworkedColor.OnValueChanged -= OnClientColorChanged;
+                }
+            }
+        }
+
+        void Update()
+        {
+            if (!IsClient)
+            {
+                return;
+            }
+
+            if (Input.GetKeyUp(KeyCode.E))
+            {
+                OnClientRequestColorChange();
```

Comment:
any reason why we're not just calling the server rpc here directly? Why the wrapper?

Reply:
same comment applies to other scripts in this project

Reply:
There are 2 reasons:

1. Architecture: when making a multiplayer game, you might need to trigger the same RPC from multiple places. Having a wrapper around makes it easy to call the RPCs by doing the proper setup/actions/local events first. It's a good practice and we should make our users aware of them, given that these samples are supposed to be copy-pastable.

2. Clarity: LOTS of users in the community are confused by how RPCs work. Having a method that starts with "OnClient" or "OnServer" makes it clear where this is originating from.

If you have further comments on the topic, please answer them in this conversation and resolve the others, so we reduce noise.

Reply:
1. ok. I'd try to show that setup then. Maybe with some RPC params. Or add some comments.
2. ok. let's make this clear with comments then